src/components/pages/locale/locale.tsx

```diff
@@ -1,71 +1,77 @@
+import { useRouter } from "next/router";
 import React from "react";
 import styled from "styled-components";
+import useSWR from "swr";
+import { v4 as uuidv4 } from "uuid";
 
-import { MagnifyingGlass } from "@src/components/icons/magnifying-glass/magnifying-glass";
+import { Button, ButtonVariant } from "@src/components/buttons/buttons";
+import { Form } from "@src/components/forms/form";
+import { FormErrorMessage } from "@src/components/input/form-error-message";
 import { InputsRadio } from "@src/components/input/input-radio-button";
-import { InputText } from "@src/components/input/input-text";
-import { deviceBreakpoints } from "@src/design-system/theme";
+import { SWRError } from "@src/errors/errors";
+import { fetchJson } from "@src/utils/fetch-json";
+import { AVAILABLE_LANGUAGE, getLocaleKey } from "@src/utils/get-locale";
 
 const Locale: React.FC = () => {
-  const availableLanguage = ["English"];
-  const [searchValue, setSearchValue] = React.useState("");
-  const [selectedLanguage, setSelectedLanguage] = React.useState<string>(
-    availableLanguage[0],
+  const router = useRouter();
+  const [formID, setFormID] = React.useState<string>(uuidv4());
+  const [locale, setLocale] = React.useState<string>("en");
+  const [errorMessage, setErrorMessage] = React.useState<string | null>(null);
+
+  const { data } = useSWR<{ locale: string }, SWRError>(
+    `/api/locale/get-locale`,
   );
 
+  React.useEffect(() => {
+    data && setLocale(data.locale);
+  }, [data]);
+
+  if (!data) {
+    return <React.Fragment />;
+  }
+
   return (
     <>
-      <SearchInputContainer>
-        <InputText
-          type="text"
-          name="search"
-          value={searchValue}
-          onChange={(value) => {
-            setSearchValue(value);
-          }}
-          label=""
-          aria-label="Search language"
-        />
-        <div className="magnifying-glass">
-          <MagnifyingGlass />
-        </div>
-      </SearchInputContainer>
+      {errorMessage ? (
+        <FormErrorMessage>{errorMessage}.</FormErrorMessage>
+      ) : null}
+      <Form
+        formID={formID}
+        onSubmit={async () => {
+          await fetchJson("/api/locale/set-locale", "POST", { locale }).then(
+            async (response) => {
+              const parsedResponse = await response.json();
+
+              if ("message" in parsedResponse) {
+                setErrorMessage("Something went wrong. Please try again later");
+                setFormID(uuidv4());
+              }
+
+              if ("updatedUser" in parsedResponse) {
+                router && router.push("/account");
+              }
 
```

Comment:
@Fenntasy Would you rather use this or the response's HTTP status code?

Reply:
Also, while working on adding an alert message, I just realized that do not return the `Item`, but this
```
{
connect-account |   '$metadata': {
connect-account |     httpStatusCode: 200,
connect-account |     requestId: '90b46643-081d-40f1-b144-6be77ca49c7c',
connect-account |     extendedRequestId: undefined,
connect-account |     cfId: undefined,
connect-account |     attempts: 1,
connect-account |     totalRetryDelay: 0
connect-account |   },
connect-account |   Attributes: undefined,
connect-account |   ConsumedCapacity: undefined,
connect-account |   ItemCollectionMetrics: undefined
connect-account | }
```